code/modules/mob/living/carbon/examine.dm

```diff
@@ -323,7 +323,7 @@
 	var/obj/item/organ/brain = get_organ_by_type(/obj/item/organ/internal/brain)
 	if(brain && HAS_TRAIT(brain, TRAIT_GHOSTROLE_ON_REVIVE))
 		return span_deadsay("[t_He] [t_is] limp and unresponsive; but [t_his] soul might yet come back...")
-	if(brain && (client || HAS_TRAIT(src, TRAIT_MIND_TEMPORARILY_GONE) || (ghost?.can_reenter_corpse && ghost?.client)))
+	if((brain && (client || HAS_TRAIT(src, TRAIT_MIND_TEMPORARILY_GONE)) || (ghost?.can_reenter_corpse && ghost?.client && (brain || !HAS_TRAIT(src, TRAIT_FAKE_SOULLESS)))))
```

Comment:
Please split this check across vars, there's 5 parenthesis at the end of this `if` it's very hard to read

Reply:
done